src/main/java/it/gov/pagopa/payhub/auth/service/AuditLoggerServiceImpl.java

```diff
@@ -0,0 +1,20 @@
+package it.gov.pagopa.payhub.auth.service;
+
+import it.gov.pagopa.payhub.auth.dto.AuditLogDTO;
+import it.gov.pagopa.payhub.auth.enums.AuditEventType;
+import it.gov.pagopa.payhub.auth.utils.AuditUtils;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j(topic = "KAFKA_LOGGER")
+public class AuditLoggerServiceImpl implements AuditLoggerService {
+  @Override
+  public void log(AuditEventType auditEventType, String mappedExternalUserId,
+      Map<String, String> label2value, String description) {
+    AuditLogDTO event = new AuditLogDTO(auditEventType, mappedExternalUserId, label2value, description);
```

Comment:
the logged user is also available through `MDC.get("externalUserId")`, which is configured when the user is correctly recognized...
what about letting the caller be unaware of keep track of it?